### DISCUSSION TITLE
Am I right to say if I don't use any client side frameworks like React, Vue, etc I should just stick to pattern-library?

### DISCUSSION THREAD

**Discussion by simkimsia** [torchbox/storybook-django#8] [Q&A] [answered]
I have been experimenting and trying to solve my own issues and after a while I am beginning to grok. 

The `storybook-django` is solely for those who want to use a client side framework like React to work with Django via REST.

If I have no intention of using a client-side and use purely serverside rendering, I should stick to pattern-library.

In other words, pattern-library is your attempt at providing a storybook for server side rendering?

Did I get that right?

**Comment by gabrielgrant** [comment]
I'd also appreciate clarification on the distinction between these two projects, and when to use one vs the other

Would also be curious to hear about how both compare to https://github.com/EmilStenstrom/django-components -- would it make sense to use either this or pattern-library to document components designed to be used with django-components?

**Comment by thibaudcolas** [comment]
FYI I’ll be working on a new version of the docs shortly. So far this has been very experimental, so spending the time writing docs didn’t feel appropriate. Now the package almost has a stable API.

For the question at hand,

- [django-pattern-library](https://github.com/torchbox/django-pattern-library) handles the rendering of Django templates with arbitrary context and tag overrides. It also provides a rudimentary UI and workflows to test templates as UI components.
- storybook-django makes it possible to test Django templates in Storybook. It relies on django-pattern-library for templates rendering only, everything else is done with Storybook APIs.
- django-components solves a different problem to the above two – which is helping you structure and reuse Django templates as components.

**Comment by simkimsia** [comment]
Looking for ward @thibaudcolas

**Comment by thibaudcolas** [comment, marked as answer]
I have updated the README with a "common questions" section that provides more details, and in particular has:

- An [explanation of how the two projects differ](https://github.com/torchbox/storybook-django/blob/main/README.md#how-does-this-compare-with-django-pattern-library)
- A [direct answer to the question of "what to use"](https://github.com/torchbox/storybook-django/blob/main/README.md#should-i-use-storybook-django-or-stick-with-django-pattern-library-only)
- A [comparison table](https://github.com/torchbox/storybook-django/blob/main/README.md#features-comparison-table)

The TL;DR; is that I would recommend storybook-django to anyone finding Storybook compelling, and django-pattern-library to anyone else who might find Storybook too complex, just wants the most basic capabilities, or wants to get started with the simplest tool possible.
